### ISSUE TITLE
Energy computation for hiking mode

### ISSUE THREAD

**Issue by Phyks** [abrensch/brouter#135]
Hi,

Following https://github.com/abrensch/brouter/commit/46c114ba1efda101d421171e27c918aaaec19758, I guess we should look at a usable energy model for hiking. So far, I found https://www.brianmac.co.uk/energyexp.htm and http://sprott.physics.wisc.edu/technote/walkrun.htm which might be of interest.

This issue is just to keep track of the ideas and possible implementations.

Best,

**Comment by abrensch** [comment]
Just a few thoughts on that:

the energy display in BRouter-Web was done originally for cars, with electric cars in mind.
The value is displayed in kWh and the meaning of the value is "mechanical energy measured at the wheel".

So if you buy a 40 kWh Renault ZOE, there's 10% "reserved" capacity and 10% in engine and batterie - so if BRouter-Web shows 32  kWh for a trip, it will fully empty the battery.

Don't know if that's a good concept - but it's exaclty the same for bikes now. At least in Germany there's a very popular category of electric bikes: engine limited to 25kmh, power limited to 250 W and usually 1/3 of total power expected from the biker itself. Battery capacities of Pedelecs are advertised in "Wh".

So here we have the situation that if BRouter-Web shows "0,40 kWh", that means pretty much that you can do that with a 400 Wh Pedelec in a medium power setting (out of low/medium/high). 400 Wh is not really what the engine brings to the wheel, but the losses here a compensated by the biker bringing it's share.

So that's a value with a real-life meaning for part of the bikers. The other part would expect the display to be in "calories" and to be equivilant to what's displayed on the ergometer in the gym. Howver, ergometers do not display mechanical energy, but multiply that by some physiiolocial factor.

So if thinking of an energy diaply for hikers I think it should also be in calories and be calibrated to what's displayd by ergometers.

**Comment by Phyks** [comment]
> the energy display in BRouter-Web was done originally for cars, with electric cars in mind.
The value is displayed in kWh and the meaning of the value is "mechanical energy measured at the wheel".

Thanks a lot for the details on the backlog and history for energy in BRouter!

> Don't know if that's a good concept - but it's exaclty the same for bikes now. At least in Germany there's a very popular category of electric bikes: engine limited to 25kmh, power limited to 250 W and usually 1/3 of total power expected from the biker itself. Battery capacities of Pedelecs are advertised in "Wh".

Same opinion here, I think this could really be a useful metrics for electric bikes (but I don't have any electric bikes nor meter to check the accuracy of BRouter computation :/). This could also be used for people doing kind of sportive biking and in need of having an estimate of burnt calories.


> So if thinking of an energy diaply for hikers I think it should also be in calories and be calibrated to what's displayd by ergometers.

Same here again, and that is what drove my research for formulas and ways to compute this :)

**Comment by 2lafru** [comment]
Hi ! I also think it would be better having energy estimation in kilo-calories. It's more suitable for someone riding a mechanical (non-electric) bike. Even if its about the net power output.

Also, in brouter-web, it would help not having to compare small fractions of KWh. Maybe the best solution is adding an unit option in brouter-web ?

**Comment by poutnikl** [comment]
Note that BRouter app and web share the common Google group, but they have independent authors and projects on GitHub. The latter had nrenner/BRouter-web, in case of posting a feature request.

**Comment by hungerburg** [comment]
Made a ride on an e-bike today: https://brouter.de/brouter-web/#map=13/47.4251/11.9295/standard&lonlats=11.880273,47.434329;11.946944,47.421554

The e-bike has an 800 Wh battery, it conforms to the standards of a pedelec (max speed, max support &c). The battery in the beginning read 99% and in in the end it read 55%, so 360 Wh consumed. Brouter tells me, the trip would require 0.25 kWh. So I consumed -110 Wh? Did I pedal backwards?

**Comment by polyscias** [comment]
> Brouter tells me, the trip would require 0.25 kWh. 

If I change in the profile the weight from 80 kg to 125 kg the energy changes from 250 Watt into 350 Watt.

Furthermore your battery consumption calculation is likely not very accurate.

The electric motor has an efficiency of ~ 90% and also not every Watt in the battery is converted to  an corresponding electric Watt because of conversion losses.

**Comment by hungerburg** [comment]
A rough computation of energy gives me 260 Wh to lift 100kg (bike+me) 935m. Obviously, brouter also accounts for some friction, a non-physical component ;) Curiously, the time spent for the journey (less than 40 minutes) gives a performance of 410W. A pedelec is limited to 250 W sustained performance, so I would have invested a staggering 130W over 40 minutes (regardless of efficiency of conversion of battery power to move us, bike and me.) Both riding and getting off the bike certainly did not feel like that.

Can this computation be made into something useful for planning trips (will the battery hold?)

PS: Would a moderator please split this topic, "Will the pedelec battery suffice?" or something along that line?

**Comment by vodie** [comment]
> A pedelec is limited to 250 W sustained performance,

No, in many countries there is no real power limit. The sustained power is the power the system can deliver without getting overheated. The manufacturer can label a 1000 W system as 250 W sustained power, because it doesn't overheat at 250 W. Many pedelecs have more than 500 W without overheating.

Your example with 100 kg and 40 minutes needs 430 W in BRouter and 0.29 kWh. You consumed 0.36 kWh at the battery with an estimated efficiency of 0.8 (controller, motor and drivetrain) this is 0.288 kWh at the wheel. With some extra losses for the intermediate/bad smoothness of the ways and maybe fat tire or low tire pressure BRouter doesn't calculate in, you have your own effort. With the right numbers BRouter is not far off.

**Comment by polyscias** [comment]
> the journey (less than 40 minutes) gives a performance of 410W. A pedelec is limited to 250 W sustained performance, so I would have invested a staggering 130W over 40 minutes

Staggering?

From [this page](https://www.fietsersbond.nl/snelheid-5-verschillende-type-fietsen-vergeleken/) under "De menselijke motor", the human engine:

- 75 watts is an effort that enables older people to meet the standard for healthy exercise and is a gentle effort for fit young people and adults.
- 100 watts is an effort that any fit adult or young person can easily maintain for an hour. 
- 150 watts is an effort that a trained cyclist can easily maintain for an hour every day. 
- Cycling at 200 watts for an hour will leave a well-trained cyclist pleasantly tired

What this already shows is that people are bad in making an estimate how much energy the deliver which makes estimates on battery life even more challenging.

**Comment by hungerburg** [comment]
Thank Y'all for the uplifting comments. I found the profile option for system weight and set "totalMass" to 100 and now I also get 280 Wh spent for the journey. I also found "bikerPower", and when I set it to "430" I get 40 minutes time to arrival :)

I also understand, 250 W is something that appears on a paper. There is one other value: The bike was equipped with a Bosch motor and in the mode used it it is specified to provide at most 340 % boost, they call it "support factor". Interpreting that like, the 430 W performed would equal 440 % system performance, i.e. 100 % from the pedals, 340 % from the motor.

This now would say, my contribution was ~100W. Further more, that would bring the system energy consumption for the trip up to 425 Wh (360 battery + 67 legs). The gap gets wider! 155 Wh (425-280) -- Could that be: 145 Wh (out of 425) for friction/distance/inefficiencies?

Can this be made into a form to get a reasonable guess when planning a trip, it the battery will last?
